pkg/handlers/generic/mutation/imageregistries/credentials/credential_provider_config_files_test.go

```diff
@@ -189,6 +190,73 @@ credentialProviders:
 `,
 			},
 		},
+
+		{
+			name:        "ECR image registry used as mirror",
+			credentials: providerConfig{URL: "https://123456789.dkr.ecr.us-east-1.amazonaws.com"},
+			mirror:      &mirrorConfig{},
+			want: &cabpkv1.File{
+				Path:        "/etc/kubernetes/dynamic-credential-provider-config.yaml",
+				Owner:       "",
+				Permissions: "0600",
+				Encoding:    "",
+				Append:      false,
+				Content: `apiVersion: credentialprovider.d2iq.com/v1alpha1
+kind: DynamicCredentialProviderConfig
+mirror:
+  endpoint: "123456789.dkr.ecr.us-east-1.amazonaws.com"
+  credentialsStrategy: "MirrorCredentialsOnly"
+credentialProviderPluginBinDir: /etc/kubernetes/image-credential-provider/
+credentialProviders:
+  apiVersion: kubelet.config.k8s.io/v1beta1
+  kind: CredentialProviderConfig
+  providers:
+  - name: ecr-credential-provider
+    args:
+    - get-credentials
+    matchImages:
+    - "123456789.dkr.ecr.us-east-1.amazonaws.com"
+    defaultCacheDuration: "0s"
+    apiVersion: credentialprovider.kubelet.k8s.io/v1alpha1
```

Comment:
This should likely be `credentialprovider.kubelet.k8s.io/v1` now, although that might only be usable if we upgrade dynamic-credential-provider image to v0.4.0

```suggestion
    apiVersion: credentialprovider.kubelet.k8s.io/v1
```

Reply:
you mean `0.5.0`? We will update it in konvoy when integrating support for image registry.

Reply:
Fixed: [546e63e](https://github.com/d2iq-labs/capi-runtime-extensions/pull/292/commits/546e63e142fc2d4f7315c47d0a9746aa2a19e034)